### PR TITLE
Rich init

### LINKED ISSUE #21: Project initilisation with metadata
As we gather a good deal of information during the project creation flow it would be convenient to pass metadata during the init of a `GitProject`.

### PR DESCRIPTION
**Pull request by kim**
Closes #21

### CODE REVIEW

File: librad/src/git/mod.rs [hunk 1]
```diff
@@ -138,35 +138,34 @@ impl GitProject {
...
-            } else {
-                src_path.file_name()
-            };
-            file_name
-                .map(|f| f.to_string_lossy().to_string())
-                .ok_or(Error::NeedProjectDir)
-        }?;
-
-        let mut pgp_key = key.clone().into_pgp(&profile.nick)?;
+        // TODO: resolve URL ref iff rad://
```
Comment (xla) [inline librad/src/git/mod.rs]:
```suggestion
        // TODO: resolve URL ref if rad://
```

Comment (kim) [reply librad/src/git/mod.rs]: `iff = if and only if` :smartass_emoji:

Comment (kim) [reply librad/src/git/mod.rs]: The reason I wrote that is that, in theory, one can also use any other URL -- but then we need to go out to the internet to fetch it. That sucks. 

Perhaps we can create a separate `Contributor` type which forces the caller to resolve the profile, and also specify if it should be stored inline or as a URL.

Comment (xla) [reply librad/src/git/mod.rs]: :exploding_head:

File: librad/src/meta/mod.rs [hunk 2]
```diff
@@ -10,3 +10,6 @@ pub use common::*;
 pub use contributor::{Contributor, ProfileRef};
 pub use profile::{Geo, ProfileImage, UserProfile};
 pub use project::{Project, Relation};
+
+pub use sequoia_rfc2822::AddrSpec;
+pub use url::Url;
```
Comment (xla) [inline librad/src/meta/mod.rs]: We also need `ParseError` to deal with issues when parsing user input.

**Pull request merged**